WarehouseManagement.Api/Controllers/VendorController.cs

```diff
@@ -0,0 +1,146 @@
+        {
+            var model = await vendorService.GetByIdAsync(id);
+            if (model == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(model);
+        }
+
+        [HttpGet("all")]
+        [ProducesResponseType(200, Type = typeof(IEnumerable<VendorDto>))]
+        public async Task<IActionResult> All()
+        {
+            var model = await vendorService.GetAllAsync();
+
+            return Ok(model);
+        }
+
+        [HttpPost("add")]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(404)]
+        [ProducesResponseType(400)]
+        public async Task<IActionResult> Add([FromBody] VendorFormDto model)
+        {
+            if (await vendorService.ExistByNameAsync(model.Name))
+            {
+                return BadRequest($"Vendor with name {model.Name} already exist");
+            }
+
+            if (await vendorService.ExistBySystemNumberAsync(model.SystemNumber))
+            {
+                return BadRequest($"Vendor with system number {model.SystemNumber} already exist");
+            }
+
+            await vendorService.AddAsync(model, User.Id());
+
+            return Ok(
+                $"Vendor with name {model.Name} and system number {model.SystemNumber} was added suscesfully"
+            );
+        }
+
+        [HttpPut("edit/{id}")]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(404)]
+        [ProducesResponseType(400)]
+        public async Task<IActionResult> Edit(int id, [FromBody] VendorFormDto model)
+        {
+            if (!await vendorService.ExistByIdAsync(id))
+            {
+                return NotFound($"Vendor with ID {id} not found.");
+            }
+
+            if (await vendorService.AnotherVendorWithNameExistAsync(id, model.Name))
+            {
+                return BadRequest($"Another vendor with name {model.Name} already exist");
+            }
+
+            if (await vendorService.AnotherVendorWithSystemNumberExistAsync(id, model.SystemNumber))
+            {
+                return BadRequest(
+                    $"Another vendor with system number {model.SystemNumber} already exist"
+                );
+            }
+
+            await vendorService.EditAsync(id, model, User.Id());
+
+            return Ok("Vendor edited successfully");
+        }
+
+        [HttpDelete("delete/{id}")]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(400)]
+        [ProducesResponseType(404)]
+        public async Task<IActionResult> Delete(int id)
+        {
+            try
+            {
+                await vendorService.DeleteAsync(id, User.Id());
+            }
+            catch (InvalidOperationException ex)
+            {
+                return BadRequest(ex.Message);
+            }
+            catch (KeyNotFoundException ex)
+            {
+                return NotFound(ex.Message);
+            }
```

Comment:
you guys need to come up with a rule when it comes to if we should write `try/catches` or for example how Vladi has made it, if checks and throw an error. For example thats how he has done it
![image](https://github.com/InterCarsBulgaria-TechLeadAcademy/warehouse.management.be/assets/18427030/a98f8d9b-d13f-411e-a438-a5e80e6d1df2)

CC: @survivorbg @SpaskoKatsarski